src/effects/backends/lv2/lv2backend.cpp

```diff
@@ -71,32 +68,30 @@ bool LV2Backend::canInstantiateEffect(const QString& effectId) const {
 }
 
 EffectManifestPointer LV2Backend::getManifest(const QString& effectId) const {
-    LV2Manifest* pLV2Mainfest = getLV2Manifest(effectId);
-    if (pLV2Mainfest != nullptr) {
-        return pLV2Mainfest->getEffectManifest();
+    VERIFY_OR_DEBUG_ASSERT(m_registeredEffects.contains(effectId)) {
+        return EffectManifestPointer();
     }
-    return EffectManifestPointer();
+    return m_registeredEffects.value(effectId);
 }
 
-LV2Manifest* LV2Backend::getLV2Manifest(const QString& effectId) const {
-    return m_registeredEffects[effectId];
+const QList<EffectManifestPointer> LV2Backend::getManifests() const {
+    QList<EffectManifestPointer> list;
+    for (const auto& manifest : m_registeredEffects) {
+        list.append(manifest);
+    }
+    return list;
 }
 
-EffectPointer LV2Backend::instantiateEffect(EffectsManager* pEffectsManager,
-                                            const QString& effectId) {
-    if (!canInstantiateEffect(effectId)) {
-        qWarning() << "WARNING: Effect" << effectId << "is not registered.";
-        return EffectPointer();
+std::unique_ptr<EffectProcessor> LV2Backend::createProcessor(
+        const EffectManifestPointer pManifest) const {
+    LV2EffectManifestPointer pLV2Manifest = m_registeredEffects.value(pManifest->id());
+    VERIFY_OR_DEBUG_ASSERT(pLV2Manifest) {
+        return std::unique_ptr<EffectProcessor>(nullptr);
```

Comment:
I think it's possible to return nullptr and unique_ptr will do the right thing:

`return nullptr;`